File: csharp/dsf/dsf/src/route/AbstractRouteStrategy.cs
namespace com.baidu.dsf.route
{

	using Service = com.baidu.dsf.register.Service;

	/// 
	/// <summary>
	/// @author arksea
	/// </summary>
	public abstract class AbstractRouteStrategy : IRouteStrategy
	{
		public abstract com.baidu.dsf.register.ServiceInstance Instance {get;}

		protected internal Service service;

		public AbstractRouteStrategy(Service svc)
		{
			this.service = svc;
		}
	}

}

File: java/dsframework/src/main/java/dsf/route/AbstractRouteStrategy.java
package dsf.route;

import dsf.register.Service;

/**
 *
 * @author arksea
 */
public abstract class AbstractRouteStrategy implements IRouteStrategy {

    protected Service service;

    public AbstractRouteStrategy(Service svc) {
        this.service = svc;
    }
}
